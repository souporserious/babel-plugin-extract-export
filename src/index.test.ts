import pluginTester from 'babel-plugin-tester'
import typescript from '@babel/plugin-syntax-typescript'

import plugin from './index'

pluginTester({
  plugin,
  pluginOptions: {
    exportIdentifierName: 'Avatar',
  },
  filename: __filename,
  snapshot: true,
  babelOptions: { plugins: [[typescript, { isTSX: true }]] },
  tests: [{ fixture: '../example.tsx' }],
})
