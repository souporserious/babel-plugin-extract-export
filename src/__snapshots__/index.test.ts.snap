// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

const unusedVariable = 'foo'

function unusedFunction() {
  return 'bar'
}

export function Avatar() {
  const unusedNestedVariable = 'foo'
  function unusedNestedFunction() {
    return 'bar'
  }
  return <img />
}

type GlobalProps = { as: any }

type BoxProps = { children: any } & GlobalProps

export const Box = (props: BoxProps) => <div {...props} />

const Stack = (props: { children: any; style?: any }) => <div {...props} />

export function Badge(props: { children: any }) {
  return <div {...props} />
}

interface ButtonProps {}

export const Button = (props: ButtonProps) => <Stack>Hello Button</Stack>

      ↓ ↓ ↓ ↓ ↓ ↓

const unusedVariable = "foo";

function unusedFunction() {
  return "bar";
}

type GlobalProps = {
  as: any;
};
type BoxProps = {
  children: any;
} & GlobalProps;
export const Box = (props: BoxProps) => <div {...props} />;


`;
